### ISSUE TITLE
[BUG] arctan2 gradient doesn't match JAX

### ISSUE THREAD

**Issue by petertsoi** [ml-explore/mlx#2451]
**Describe the bug**
The gradients I get from `jnp.arctan2` are different from what I get from `mx.arctan2`.

**To Reproduce**

```
import jax
import jax.numpy as jnp

import mlx.core as mx

x1 = [1.0, 2.0, 3.0, 4.0]
x2 = [2.0, 3.0, 4.0, 5.0]

x1_jax = jnp.array(x1)
x2_jax = jnp.array(x2)

x1_mx = mx.array(x1)
x2_mx = mx.array(x2)

jax_value, jax_grad = jax.vmap(jax.value_and_grad(jnp.arctan2))(x1_jax, x2_jax)

mx_value = mx.vmap(mx.arctan2)(x1_mx, x2_mx)
mx_grad = mx.vmap(mx.grad(mx.arctan2))(x1_mx, x2_mx)

print("Values appear similar:")
print(jax_value)
print(mx_value)

print("Grads have non-negligible differences:")
print(jax_grad)
print(mx_grad)
```

**Expected behavior**
```
Values appear similar:
[0.4636476  0.5880026  0.6435011  0.67474097]
array([0.463648, 0.588003, 0.643501, 0.674741], dtype=float32)

Grads have non-negligible differences:
[0.4        0.23076923 0.16       0.12195122]
array([0.2, 0.0769231, 0.04, 0.0243902], dtype=float32)
```

**Desktop**
 - macOS: 15.5
 - MLX: 0.27.1
 - Python 3.11.13

**Comment by virtan** [comment]
I wasn't able to check the fix, but it is possible that Jacobian for arctan2 has a bug in the implementation. This may fix the issue:
```
--- a/mlx/primitives.cpp
+++ b/mlx/primitives.cpp
@@ -518,12 +518,19 @@ std::vector<array> ArcTan2::jvp(
     const std::vector<array>& tangents,
     const std::vector<int>& argnums) {
   assert(primals.size() == 2);
+  assert(tangents.size() == 2);
   assert(argnums.size() == 2);
-  array t =
-      add(square(primals[0], stream()), square(primals[1], stream()), stream());
-  return {
-      divide(tangents[0], t, stream()),
-      divide(negative(tangents[1], stream()), t, stream())};
+
+  const array& y = primals[0];
+  const array& x = primals[1];
+  const array& dy = tangents[0];
+  const array& dx = tangents[1];
+
+  array t = add(square(x, stream()), square(y, stream()), stream());
+  array num = subtract(multiply(x, dy, stream()), multiply(y, dx, stream()), stream());
+  array jvp = divide(num, t, stream());
+
+  return {jvp};
 }
```
